File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/CtypeCntrl.java
package com.onevgo.functions;

import java.util.LinkedHashMap;
import java.util.Map;

public class CtypeCntrl {
    public static boolean ctypeCntrl(String s) {
        if (s == null || s.length() == 0) return false;
        for (int i = 0; i < s.length(); i++) {
            char codePoint = s.charAt(i);
            // 0-31 or 127
            if ((codePoint >= 0 && codePoint <= 31) ||
                    codePoint == 127) {
                continue;
            }
            return false;
        }
        return true;
    }

    public static void main(String[] args) {
        Map<String, String> map = new LinkedHashMap<>();
        map.put("string1", "\n\r\t");
        map.put("string2", "arf12");
        map.put("", "");
        for (Map.Entry<String, String> entry : map.entrySet()) {
            if (ctypeCntrl(entry.getValue())) {
                System.out.println("The string " + entry.getKey() + " consists of all control characters.");
            } else {
                System.out.println("The string " + entry.getKey() + " does not consist of all control characters.");
            }
        }
    }
}

File: 01_Language/01_Functions/js/ctype_cntrl.js
function ctype_cntrl(text) {
    if (text === null || text === undefined || text === '') return false
    for (let i in text) {
        let codePoint = text.charCodeAt(i)
        if ((codePoint >= 0 && codePoint <= 31) ||
            codePoint === 127) {
            continue
        }
        return false
    }
    return true
}

let strings = {'string1': "\n\r\t", 'string2': 'arf12', '': ''}
for (let k in strings) {
    if (ctype_cntrl(strings[k])) {
        console.log("The string '" + k + "' consists of all control characters.");
    } else {
        console.log("The string '" + k + "' does not consist of all control characters.");
    }
}
